influunt-api/app/models/Controlador.java

```diff
@@ -101,7 +101,7 @@
     @PrivateOwned
     private List<Anel> aneis;
 
-    @OneToMany(mappedBy = "controlador")
+    @OneToMany(mappedBy = "controlador", cascade = CascadeType.ALL)
```

Comment:
Se o cascade foi colocado aqui somente para deletar o grupo semafórico, recomendo usar o `CascadeType.REMOVE`. Mas parece que tá sendo chamado o `delete()` direto no grupo semafórico, então acho que esse cascade nem é necessário (se o grupo semafórico for salvo através do cascade do anel).